src/python/pants/option/global_options.py

```diff
@@ -421,7 +422,9 @@ def from_options(cls, options: OptionValueContainer) -> LocalStoreOptions:
     process_execution_local_cache=True,
     # Remote store setup.
     remote_store_address=None,
-    remote_store_headers={},
+    remote_store_headers={
+        "user-agent": f"pants/{VERSION}",
```

Comment:
should be capitalized: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent

Reply:
Are you sure? Curl doesn't capitalize: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent#examples_7

Reply:
Not for HTTP/2 and thus not for gRPC. All HTTP/2 headers are lower case.

> Just as in HTTP/1.x, header field names are strings of ASCII characters that are compared in a case-insensitive fashion.  However, header field names MUST be converted to lowercase prior to their encoding in HTTP/2.  A request or response containing uppercase header field names MUST be treated as malformed (Section 8.1.2.6).

https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2

Reply:
We should probably normalize `--remote-store-headers` and `--remote-execution-headers` to ensure all header names are lower case. Even though `tonic` and `hyper` crates will do that for us, the Rust code I wrote to extract `user-agent` expects it to be lower case (as per the standard).